staging/src/k8s.io/apiserver/pkg/storage/cacher/cache_watcher.go

```diff
@@ -392,9 +392,9 @@ func (c *cacheWatcher) convertToWatchEvent(event *watchCacheEvent) *watch.Event
 
 	switch {
 	case curObjPasses && !oldObjPasses:
-		return &watch.Event{Type: watch.Added, Object: getMutableObject(event.Object)}
+		return &watch.Event{Type: watch.Added, Object: event.Object}
```

Comment:
in case we would like to have more soak time, we could put that change behind the feature gate (WatchList)